src/components/resource-picker/media-library/media-library-search-screen.js

```diff
@@ -0,0 +1,193 @@
+import { Alert, Button, Spin } from 'antd';
+import reactDropzoneNs from 'react-dropzone';
+import NoSearch from '../shared/no-search.js';
+import { useUser } from '../../user-context.js';
+import FilesGridViewer from '../files-grid-viewer.js';
+import { Trans, useTranslation } from 'react-i18next';
+import React, { Fragment, useRef, useState } from 'react';
+import { SOURCE_TYPE } from '../../../domain/constants.js';
+import UploadIcon from '../../icons/general/upload-icon.js';
+import ResourceSearchBar from '../shared/resource-search-bar.js';
+import permissions, { hasUserPermission } from '../../../domain/permissions.js';
+import { mediaLibraryItemWithRelevanceShape } from '../../../ui/default-prop-types.js';
+import { ALLOWED_MEDIA_LIBRARY_RESOURCE_TYPES } from '../../../utils/media-library-utils.js';
+
+const ReactDropzone = reactDropzoneNs.default || reactDropzoneNs;
+
+const SCREEN = {
+  search: 'search',
+  searchInvitation: 'search-invitation',
+  initialUrlPreview: 'initial-url-preview'
+};
+
+function MediaLibrarySearchScreen({
+  files,
+  isHidden,
+  isLoading,
+  initialUrl,
+  searchParams,
+  highlightedFile,
+  onFileDrop,
+  onFileClick,
+  onCancelClick,
+  onFileDoubleClick,
+  onDeleteFileClick,
+  onPreviewFileClick,
+  onSearchParamsChange,
+  onSelectInitialUrlClick,
+  onSelectHighlightedFileClick
+}) {
+  const user = useUser();
+  const dropzoneRef = useRef();
+  const { t } = useTranslation('mediaLibrarySearchScreen');
+  const [hasSearchedAtLeastOnce, setHasSearchedAtLeastOnce] = useState(false);
+
+  let currentScreen;
+  let canSelectUrl;
+  if (hasSearchedAtLeastOnce) {
+    currentScreen = SCREEN.search;
+    canSelectUrl = !!highlightedFile;
+  } else if (initialUrl) {
+    currentScreen = SCREEN.initialUrlPreview;
+    canSelectUrl = true;
+  } else {
+    currentScreen = SCREEN.searchInvitation;
+    canSelectUrl = false;
+  }
+
+  const handleSearch = newSearchParams => {
+    setHasSearchedAtLeastOnce(true);
+    onSearchParamsChange(newSearchParams);
+  };
+
+  const handleSelectClick = () => {
+    switch (currentScreen) {
+      case SCREEN.search:
+        return onSelectHighlightedFileClick();
+      case SCREEN.initialUrlPreview:
+        return onSelectInitialUrlClick();
+      default:
+        throw new Error('No file selected');
+    }
+  };
+
+  const handleUploadButtonClick = () => {
+    dropzoneRef.current.open();
+  };
+
+  const renderSearchInfo = () => {
+    let searchMessage;
+    if (isLoading) {
+      searchMessage = t('common:searchOngoing');
+    } else if (searchParams.searchTerm) {
+      searchMessage = (
+        <Trans
+          t={t}
+          i18nKey="common:searchResultInfo"
+          values={{ resultCount: files.length, searchTerm: searchParams.searchTerm }}
+          components={[<i key="0" />]}
+          />
+      );
+    } else {
+      searchMessage = null;
+    }
+
+    return searchMessage ? <Alert type="info" message={searchMessage} showIcon /> : null;
```

Comment:
Tiny tiny thing, but I think direct returns of `<Alert ..>` elements or `null` at the end would be nicer and simpler than `if-else-else` here. It's just one repetition of a straight forward element.